### ISSUE TITLE
Raycast and pointer enter/leave handling

### ISSUE THREAD

**Issue by SaFrMo** [troisjs/trois#31]
While working with #30, I was wondering if there's a cleaner way to handle raycasting and pointer events. I'm thinking of a two-step approach, raycaster-first and object-first.

### Raycaster-first

Sometimes I want to raycast against everything in a crowded scene - in this case, I'd rather write a single raycaster's `pointerEnter` and `pointerLeave` functions than those of all the objects in the scene. To accomplish this, we could add a `Raycaster` component:

```html
<Camera>
  <Raycaster @pointerEnter="onEnter" @pointerLeave="onLeave"/>
</Camera>
```

```js
export default {
  methods: {
    onEnter(intersections) {
      console.log(intersections)
    },
    onLeave(inactiveIntersections){ 
      console.log(inactiveIntersections)
    }
  }
}
```

This would raycast against everything (`raycaster.intersectObjects(scene.children)`) and fire appropriate callbacks when pointer interactions start and end. We could also include an optional prop `targetObjects` if the user wants to specify which objects to target if they don't want all of them.

### Object-first

I'm finding the `<Renderer mouse-over>` and mesh `:onHover` prop a little confusing - maybe we could do something like this:

* Add `pointerEnter` and `pointerLeave` props to all Meshes
* In a mesh's initialization, if either `pointerEnter` or `pointerLeave` are defined, check for  `this.three.raycaster`. 
  * If one is defined, add the mesh to the list of target objects the raycaster can intersect.
  * If one is not defined, tell `this.three` to create one (a new method like `this.three.initializeRaycaster()`), then add the mesh to the list of target objects.

From there, `three.raycaster` can handle calling events on meshes when the pointer enters or leaves them. This way we don't need any `Renderer`-level awareness of raycasting and the `pointerEnter`/`pointerLeave` events more closely match the Raycast component above (plus they're more clearly named for handling non-mouse inputs).

Happy to add this in the next couple days if it sounds worthwhile - let me know any thoughts @klevron !

**Comment by klevron** [comment]
Thanks, sounds great, I was thinking the same about replacing mouse with pointer

**Comment by SaFrMo** [comment]
Fixed in #33
